gateway/ln-gateway/src/gatewayd/actor.rs

```diff
@@ -57,7 +75,114 @@ impl GatewayActor {
             }
         });
 
-        Ok(Self { client })
+        let actor = Self {
+            client,
+            lnrpc,
+            task_group,
+        };
+
+        actor.subscribe_htlcs().await?;
```

Comment:
(just writing out thoughts) 
I wonder if we should separate the `new()` and the `subscribe_htlcs()` method.  

`new` just creates registered GatewayActor and returns it and then we would call `.subscribe_htlcs` on it. 
However, it is "one step" and would require to make the `subscribe_htlcs` method public which we might not want to do. 

opinion?

Reply:
At this moment, I think an actor is useless without an active subscription for htlcs. For this reason, `subscribe_htlcs` made sense an important pre-condition to bootstrapping and returning a new, valid actor.

We can make it one step at a later time, when we have reasons for actors to be valid without active subs for htlcs.

Reply:
sounds reasonable :+1: